### ISSUE TITLE
Pin versions

### ISSUE THREAD

**Issue by manoadamro** [OlafenwaMoses/ImageAI#533]
ImageAI clearly needs specific versions for tensorflow, keras etc. I have spent 2 days, going through various conda environments, trying to figure out which specific combination of tensorflow-gpu/keras/cuda/cudnn I need with no luck.

Would it be possible to pin the versions in requirements.txt?

It would also be helpful to know which versions of cuda/cudnn are required for which versions of tensorflow-gpu since again, after 2 days, I can't find a single combination that works.

If anyone has managed to sucessfully use this with gpu (or even cpu at this point), using conda, please let me know how you did it.

**Comment by inquirer5** [comment]
I feel your pain as I just struggled through this tonight as well and I had to use Conda since I'm running this on a machine without the Intel AVX extensions. Here is the recipe that worked for me using Nvidia GPUs on Ubuntu 18.04:
Note: For whatever reason that opencv version won't install for me with conda anymore and I had to use pip from my conda env for that one.
```
tensorflow-gpu==1.13.1
opencv-python==4.2.0.32
keras==2.3.1
```

For CUDA I followed the Tensorflow-GPU instructions [here](https://www.tensorflow.org/install/gpu#ubuntu_1804_cuda_101), but switched to the following versions:
Note the CUDA 10.0 deb can be found [here](https://developer.nvidia.com/cuda-10.0-download-archive?target_os=Linux&target_arch=x86_64&target_distro=Ubuntu&target_version=1804&target_type=debnetwork)
```
sudo apt-get install -y --no-install-recommends cuda-10-0 libcudnn7=7.3.1.20-1+cuda10.0 libcudnn7-dev=7.3.1.20-1+cuda10.0
sudo apt-get install -y --no-install-recommends libnvinfer6=6.0.1-1+cuda10.0 libnvinfer-dev=6.0.1-1+cuda10.0 libnvinfer-plugin6=6.0.1-1+cuda10.0
```

Hope that helps.
